src/json-rpc-provider-types.test-d.ts

```diff
@@ -0,0 +1,34 @@
+import { Web3Provider } from '@ethersproject/providers';
+import EthQuery from '@metamask/eth-query';
+import { expectAssignable, expectNotAssignable } from 'tsd';
+
+import type { JsonRpcRequest, LegacyEthereumProvider } from '.';
+
+// Known legacy providers
+expectAssignable<LegacyEthereumProvider>(new EthQuery({} as any));
```

Comment:
```suggestion
// @ts-expect-error TODO: align `EverythingButNull` type in `@metamask/eth-query`
expectAssignable<LegacyEthereumProvider>(new EthQuery({} as any));
```

- This error results from the `EverythingButNull` typing update in [7b176aa](https://github.com/MetaMask/utils/pull/140/commits/7b176aa64826d7a5fe9bbdea9db32786668e55cf).
- `NonNullable<unknown> | undefined` or `{} | undefined` is the more "correct" typing for `EverythingButNull` than the enumeration of basic types.
- `@metamask/eth-query` has a duplicate type definition for `EverythingButNull`, which should be replaced with an import from `utils`. To achieve this, the type definition in this repo should be exported and probably moved to `misc.ts`.